Source/Eto.WinForms/Forms/WindowsControl.cs

```diff
@@ -91,6 +93,79 @@ public abstract class WindowsControl<TControl, TWidget, TCallback> : WidgetHandl
 		where TWidget : Control
 		where TCallback : Control.ICallback
 	{
+
+		// used in DrawableHandler
+		public class PanelBase<THandler> : swf.Panel
+			where THandler: WindowsControl<TControl, TWidget, TCallback>
+		{
+			public THandler Handler { get; set; }
+
+			public PanelBase( THandler handler = null )
+			{
+				Handler = handler;
+				Size = sd.Size.Empty;
+				MinimumSize = sd.Size.Empty;
+				AutoSize = true;
+				AutoSizeMode = swf.AutoSizeMode.GrowAndShrink;
+			}
+
+			public override sd.Size GetPreferredSize(sd.Size proposedSize)
+			{
+				var userSize = Handler.UserDesiredSize;
+				var size = userSize.Width >= 0 && userSize.Height >= 0 ? sd.Size.Empty
+					: base.GetPreferredSize(proposedSize);
+				if (userSize.Width >= 0)
+					size.Width = Math.Max(userSize.Width, MinimumSize.Width);
+				if (userSize.Height >= 0)
+					size.Height = Math.Max(userSize.Height, MinimumSize.Height);
+				return size;
+			}
+			// Need to override IsInputKey to capture 
+			// the arrow keys.
+			protected override bool IsInputKey(swf.Keys keyData)
+			{
+				switch (keyData & swf.Keys.KeyCode)
+				{
+					case swf.Keys.Up:
+					case swf.Keys.Down:
+					case swf.Keys.Left:
+					case swf.Keys.Right:
+					case swf.Keys.Back:
+						return true;
+					default:
+						return base.IsInputKey(keyData);
+				}
+			}
+		}
+
+		// used in Panel+PixelLayout and similar code is in TableLayout
+		public class EtoPanel<THandler> : PanelBase<THandler>
+			where THandler : WindowsControl<TControl, TWidget, TCallback>
+		{
+			public EtoPanel( THandler handler = null )
+				: base( handler )
+			{ }
+
+			// optimization especially for content on drawable
+			protected override void OnBackColorChanged( EventArgs e )
+			{
+				SetStyle
+					( swf.ControlStyles.AllPaintingInWmPaint
+					| swf.ControlStyles.DoubleBuffer
+					, BackColor.A != 255 );
+				base.OnBackColorChanged( e );
+			}
+			protected override void OnParentBackColorChanged( EventArgs e )
+			{
+				SetStyle
+					( swf.ControlStyles.AllPaintingInWmPaint
+					| swf.ControlStyles.DoubleBuffer
+					, BackColor.A != 255 );
+				base.OnParentBackColorChanged( e );
+			}
+		}
```

Comment:
This eliminates flicker for content on drawable
...and generally transparent background